scripts/mempool.js

```diff
@@ -0,0 +1,248 @@
+    static PENDING = 2;
+
+    /** The DELEGATED PENDING state (cold UTXO is in mempool, pending confirmation) */
+    static PENDING_COLD = 3;
+
+    /** The CONFIRMED DELEGATED state (cold UTXO is spendable) */
+    static DELEGATE = 4;
+
+    /** The REWARD state (UTXO is a reward from a stake or masternode) */
+    static REWARD = 5;
+
+    /** The block count, updated automatically by the network sync module
+     * @type {Number}
+     */
+    static blockCount;
+
+    /**
+     * Remove a UTXO after a set amount of time
+     * @param {Number} nBlocks - Estimated blocks to wait
+     * @param {UTXO} cUTXO - UTXO to remove
+     */
+     async removeWithDelay(nBlocks, cUTXO) {
+        await sleep(nBlocks * 60 * 1000);
+        this.removeUTXO(cUTXO);
+    }
+
+    /**
+     * Check if an exact UTXO match can be found in our wallet
+     * @param {UTXO} cNewUTXO - UTXO to search for
+     * @returns {Boolean} `true` or `false`
+     */
+    isAlreadyStored(cNewUTXO) {
+        for (const cUTXO of this.UTXOs) {
+            if (JSON.stringify(cUTXO) === JSON.stringify(cNewUTXO)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Fetches an array of UTXOs filtered by their state
+     * @param {Number} nState - Specific UTXO state
+     * @returns {Array<UTXO>} `array` - An array of UTXOs
+     */
+    getUTXOsByState(nState) {
+        return this.UTXOs.filter(cUTXO => cUTXO.status === nState);
+    }
+
+    /**
+     * Fetches an array of confirmed UTXOs, an easier alias to {@link getUTXOsByState}
+     * @returns {Array<UTXO>} `array` - An array of UTXOs
+     */
+    getConfirmed() {
+        return this.getUTXOsByState(Mempool.CONFIRMED);
+    }
+
+    /**
+     * Removes a UTXO from a specific state
+     * @param {UTXO} cNewUTXO - Pending UTXO to remove
+     * @param {Number} nState - Specific state of this UTXO to search for
+     */
+    removeFromState(cNewUTXO, nState) {
+        const arrPendingUTXOs = this.getUTXOsByState(nState);
+        // Loop each pending UTXO
+        for (const cUTXO of arrPendingUTXOs) {
+            // Search for matching ID + output number
+            if (cUTXO.id === cNewUTXO.id && cUTXO.vout === cNewUTXO.vout) {
+                // Nuke it from orbit
+                this.removeUTXO(cUTXO);
+                break;
+            }
+        }
+    }
+
+    /**
+     * Add a new UTXO to the wallet
+     * @param {String} id - Transaction ID
+     * @param {String} path - If applicable, the HD Path of the owning address
+     * @param {Number} sats - Satoshi value in this UTXO
+     * @param {String} script - HEX encoded spending script
+     * @param {Number} vout - Output position of this transaction
+     * @param {Number} height - Block height of the UTXO
+     * @param {Number} status - UTXO status enum state
+     */
+    addUTXO(id, path, sats, script, vout, height, status) {
```

Comment:
Style nit: Using a deconstructed object would be more readable, lines like this `mempool.addUTXO(futureTxid,"",nChange,"",0,"",Mempool.PENDING_COLD);` Could be changed to: `mempool.addUTXO({id: futureTxid, sats: nChange, vout: 0, status: Mempool.PENDING_COLD})`

Reply:
fixed